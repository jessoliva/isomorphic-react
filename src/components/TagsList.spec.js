// anywhere you use JSX, you need to import React
import React from 'react';

// import component itself
import TagsList from './TagsList'

// to render the component
import renderer from 'react-test-renderer';


describe("The tags list",()=>{
    /**
     * The tagsList can be tested against an expected snapshot value, as in below.
     */
    it ("renders as expected",()=>{

        // define tree
        // a JSON representation of the component
        // pass a react component to renderer.create and it returns a json representation of the component
        const tree = renderer
           .create(<TagsList tags={[`css`,`go`,`typescript`,`coffeescript`]}/>)
           .toJSON();
           // call to JSON so that it returns a JSON representation of the component

        // compare tree to snapshot
        expect(tree).toMatchSnapshot();
    });  
});