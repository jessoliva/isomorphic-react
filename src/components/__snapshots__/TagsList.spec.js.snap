// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The tags list renders as expected 1`] = `
<div>
  <code>
    css
  </code>
  <code>
    go
  </code>
  <code>
    typescript
  </code>
  <code>
    coffeescript
  </code>
</div>
`;
